privacy-on-beam/pbeam/aggregations.go

```diff
@@ -450,6 +483,156 @@ func clampNegativePartitionsFloat64Fn(v beam.V, r float64) (beam.V, float64) {
+	}
+	return fn
+}
+
+func prepareAddPartitionsInt64Fn(partitionKey beam.X) (k beam.X, v int64) {
+	return partitionKey, 0
+}
+
+func prepareAddPartitionsFloat64Fn(partitionKey beam.X) (k beam.X, v float64) {
+	return partitionKey, 0
+}
+
+func prepareAddPartitionsMeanInt64Fn(partitionKey beam.X) (k beam.X, v []int64) {
+	return partitionKey, []int64{}
+}
+
+func prepareAddPartitionsMeanFloat64Fn(partitionKey beam.X) (k beam.X, v []float64) {
+	return partitionKey, []float64{}
+}
+
+// Function for sum and mean. Drop unspecified partitions.
+func dropUnspecifiedPartitions(s beam.Scope, partitionsCol beam.PCollection, pcol PrivatePCollection, partitionEncodedType beam.EncodedType) beam.PCollection {
+	partitionMap := beam.Combine(s, newPartitionsHashMapFn(partitionEncodedType), partitionsCol)
+	return beam.ParDo(s, prunePartitionsFn, pcol.col, beam.SideInput{Input: partitionMap})
+
+}
+
+// Function for count and distinct_id. Drop unspecified partitions.
+func dropUnspecifiedPartitionsForCount(s beam.Scope, partitionsCol beam.PCollection, pcol PrivatePCollection, partitionFullType typex.FullType) beam.PCollection {
+	partitionEncodedType := beam.EncodedType{partitionFullType.Type()}
+	partitionMap := beam.Combine(s, newPartitionsHashMapFn(partitionEncodedType), partitionsCol)
+	return beam.ParDo(s, newPrunePartitionsFnForCount(partitionFullType), pcol.col, beam.SideInput{Input: partitionMap})
+}
+
+type mapAccum struct {
+	// Key is string representation of partition encoding.
+	// Value is true.
+	PartitionMap PMap
+}
+
+// Make hashmaps for partitions.
+type PartitionsHashMapFn struct {
+	PartitionType beam.EncodedType
+	partitionEnc  beam.ElementEncoder
+}
+
+func newPartitionsHashMapFn(partitionType beam.EncodedType) *PartitionsHashMapFn {
+	return &PartitionsHashMapFn{
+		PartitionType: partitionType,
+	}
+}
+
+func (fn *PartitionsHashMapFn) Setup() {
+	fn.partitionEnc = beam.NewElementEncoder(fn.PartitionType.T)
+}
+
+func (fn *PartitionsHashMapFn) CreateAccumulator() mapAccum {
+	return mapAccum{PartitionMap: make(PMap)}
+}
+
+func (fn *PartitionsHashMapFn) AddInput(m mapAccum, partitionKey beam.X) mapAccum {
+	var partitionBuf bytes.Buffer
+	if err := fn.partitionEnc.Encode(partitionKey, &partitionBuf); err != nil {
+		log.Exitf("pbeam.PartitionsHashMapFn.ProcessElement: couldn't encode partition key %v: %v", partitionKey, err)
+	}
+	encodedPartitionKey := string(partitionBuf.Bytes())
+	m.PartitionMap[encodedPartitionKey] = true
+	return m
+}
+
+func (fn *PartitionsHashMapFn) MergeAccumulators(a, b mapAccum) mapAccum {
+	for k := range a.PartitionMap {
+		b.PartitionMap[k] = true
+	}
+	return b
+}
+
+func (fn *PartitionsHashMapFn) ExtractOutput(m mapAccum) PMap {
+	return m.PartitionMap
+}
+
+// prunePartitionsFnForCount takes a PCollection<K, V> as input, and returns a
+// PCollection<K, V>, where unspecified partitions have been dropped.
+// Used for count and distinct_id.
+type prunePartitionsFnForCount struct {
```

Comment:
The same naming suggestion here with `prunePartitionsVFn` and `prunePartitionsKVFn`